xml_analyzer/data_container.py

```diff
@@ -6,21 +6,31 @@ def __init__(self):
         self.mapping_element = collections.namedtuple('property_value_mapping', ['property', 'value'])
         self.mappings = dict()
         self.mappings['variables'] = dict()
+        self.mappings['autocalc'] = dict()
         self.result_property = ''
         self.dependencies = list()
         self.constraint_specification = ''
 
     def add_prop_val_mapping(self, prop: str, value: str):
-        map_element = self.mapping_element(prop, value)
-        self.mappings['variables'][prop] = value
+        self.mappings['variables'][prop] = float(value)
 
     def add_result_property(self, prop):
```

Comment:
type hint

Reply:
Also have a look into @ property this is what you want here

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/6614ef2d31e5e3284ee434bbd113e446107a9606. Great hint!!